javalin/src/main/java/io/javalin/core/util/HttpAllowedMethodsOnRoutesUtil.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Javalin - https://javalin.io
+ * Copyright 2021 David Åse
+ * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
+ */
+
+package io.javalin.core.util
+
+import io.javalin.Javalin
+import io.javalin.core.event.HandlerMetaInfo
+import io.javalin.core.plugin.Plugin
+import io.javalin.core.plugin.PluginLifecycleInit
+
+class HttpAllowedMethodsOnRoutesUtil() : Plugin, PluginLifecycleInit {
+    private val endpoints = mutableMapOf<String, MutableSet<HandlerMetaInfo>>()
+
+    override fun init(app: Javalin) {
+        app.events {
+            it.handlerAdded { handlerInfo ->
+                addOptionsToList(handlerInfo)
+            }
+        }
+    }
+
+    override fun apply(app: Javalin) {
+        app.events {
+            it.serverStarted {
+                createOptionsEndPoint(app)
+            }
+        }
+    }
+
+    private fun addOptionsToList(handlerMetaInfo: HandlerMetaInfo) {
+        if (endpoints[handlerMetaInfo.path] == null) {
+            endpoints[handlerMetaInfo.path] = mutableSetOf(handlerMetaInfo)
+        } else {
+            endpoints[handlerMetaInfo.path]!!.add(handlerMetaInfo)
+        }
+    }
+
+    private fun createOptionsEndPoint(app: Javalin) {
+        endpoints.forEach { endpoint ->
+            app.options(
+                endpoint.key
```

Comment:
What is the purpose of keeping this as a map, rather than just keeping it as a `Set<HandlerMetaInfo>`?

Reply:
I guess it's more efficient than filtering every time 😄